java/src/gov/usgs/producerclient/ProducerClient.java

```diff
@@ -0,0 +1,313 @@
+
+import org.apache.log4j.Logger;
+import org.apache.log4j.BasicConfigurator;
+import org.apache.log4j.PropertyConfigurator;
+
+import gov.usgs.hazdevbroker.Producer;
+
+import java.util.*;
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.IOException;
+
+import org.json.simple.JSONObject;
+import org.json.simple.parser.JSONParser;
+import org.json.simple.parser.ParseException;
+
+/**
+ * a client class used to produce messages to a hazdev-broker (kafka) topic and
+ * from messages in files based on the provided configuration
+ *
+ * @author U.S. Geological Survey &lt;jpatton at usgs.gov&gt;
+ */
+public class ProducerClient {
+	/**
+	 * JSON Configuration Keys
+	 */
+	public static final String LOG4J_CONFIGFILE = "Log4JConfigFile";
+	public static final String BROKER_CONFIG = "HazdevBrokerConfig";
+	public static final String TOPIC = "Topic";
+	public static final String FILE_EXTENSION = "FileExtension";
+	public static final String TIME_PER_FILE = "TimePerFile";
+	public static final String INPUT_DIRECTORY = "InputDirectory";
+	public static final String ARCHIVE_DIRECTORY = "ArchiveDirectory";
+
+	/**
+	 * Required configuration string defining the input directory
+	 */
+	private static String inputDirectory;
+
+	/**
+	 * Optional configuration string defining the archive directory
+	 */
+	private static String archiveDirectory;
+
+	/**
+	 * Required configuration string defining the input file extension
+	 */
+	private static String fileExtension;
+
+	/**
+	 * Optional configuration Long defining the number seconds before reading a
+	 * file, default is null
+	 */
+	private static Long timePerFile;
+
+	/**
+	 * Log4J logger for ProducerClient
+	 */
+	static Logger logger = Logger.getLogger(ProducerClient.class);
+
+	/**
+	 * main function for ConsumerClient
+	 *
+	 * @param args
+	 *            - A String[] containing the command line arguments.
+	 */
+	public static void main(String[] args) {
+
+		// check number of arguments
+		if (args.length == 0) {
+			System.out.println(
+					"Usage: hazdev-broker ProducerClient <configfile>");
+			System.exit(1);
+		}
+
+		// init to default values
+		inputDirectory = null;
+		archiveDirectory = null;
+		fileExtension = null;
+		timePerFile = null;
+
+		// get config file name
+		String configFileName = args[0];
+
+		// read the config file
+		File configFile = new File(configFileName);
+		BufferedReader configReader = null;
+		StringBuffer configBuffer = new StringBuffer();
+		try {
+			configReader = new BufferedReader(new FileReader(configFile));
+			String text = null;
+
+			while ((text = configReader.readLine()) != null) {
+				configBuffer.append(text).append("\n");
+			}
+		} catch (FileNotFoundException e) {
+			e.printStackTrace();
+		} catch (IOException e) {
+			e.printStackTrace();
+		} finally {
+			try {
+				if (configReader != null) {
+					configReader.close();
+				}
+			} catch (IOException e) {
```

Comment:
probably should not have an empty catch block

Reply:
Addressed in new commit.